File: rust/axum-postgres/src/handlers/health.rs
use axum::{extract::State, http::StatusCode, Json};
use serde_json::{json, Value};
use sqlx::Row;

use crate::AppState;

pub async fn health_check(State(state): State<AppState>) -> (StatusCode, Json<Value>) {
    let db_status = sqlx::query("SELECT 1 as one")
        .fetch_one(&state.pool)
        .await
        .map(|row: sqlx::postgres::PgRow| {
            let _: i32 = row.get("one");
            "healthy"
        })
        .unwrap_or("unhealthy");

    let status = if db_status == "healthy" {
        StatusCode::OK
    } else {
        StatusCode::SERVICE_UNAVAILABLE
    };

    (
        status,
        Json(json!({
            "status": if status == StatusCode::OK { "ok" } else { "error" },
            "database": db_status,
            "service": "rust-axum-postgres",
        })),
    )
}

File: nodejs/ai-contract-analyzer/src/routes/health.ts
import { Hono } from "hono";
import { getPool } from "../db/pool.ts";

const health = new Hono();

health.get("/health", async (c) => {
  const pool = getPool();
  try {
    await pool.query("SELECT 1");
    return c.json({ status: "ok", db: "connected" });
  } catch {
    return c.json({ status: "error", db: "disconnected" }, 503);
  }
});

export { health };
